### PR TITLE
refactored ldap utils

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@liggitt  @deads2k This is the refactor of the utils for Jordan's `ldappassword` code

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
let's [test]

**Comment by stevekuznetsov** [comment]
`test-cmd` flake (`Error from server: pods "hello-openshift" cannot be updated: the object has been modified; please apply your changes to the latest version and try again`), re[test]

**Comment by deads2k** [comment]
https://github.com/openshift/origin/issues/4262

**Comment by stevekuznetsov** [comment]
Unrelated Jenkins flake:
```
[INFO] Waiting for Docker registry pod to start
[INFO] Waiting for command to finish: '[[ "$(oc get endpoints docker-registry --output-version=v1beta3 -t "{{ if .subsets }}{{ len .subsets }}{{ else }}0{{ end }}" || echo "0")" != "0" ]]'...
[ ERR] Gave up waiting for: '[[ "$(oc get endpoints docker-registry --output-version=v1beta3 -t "{{ if .subsets }}{{ len .subsets }}{{ else }}0{{ end }}" || echo "0")" != "0" ]]'
[FAIL] !!!!! Test Failed !!!!
```

File: pkg/auth/authenticator/password/ldappassword/ldaputil/getattribute.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"strings"
+
+	"github.com/go-ldap/ldap"
+)
+
+// GetAttributeValue finds the first attribute of those given that the LDAP entry has, and
+// returns it. GetAttributeValue is able to query the DN as well as Attributes of the LDAP entry.
+// If no value is found, the empty string is returned.
+func GetAttributeValue(entry *ldap.Entry, attributes []string) string {
```
Comment (deads2k) [inline pkg/auth/authenticator/password/ldappassword/ldaputil/getattribute.go]: Move closer to usage.  I don't see much value in giving this its own file.

**Comment by deads2k** [comment]
@liggitt I didn't inspect very deeply, but this looks like a pretty straight shuffle of methods to prep us for ldap sync.  I'm inclined to take it.

One thing I don't like is having the package under `pkg/auth/authenticator/password/ldappassword/ldaputil`.  `pkg/auth/ldaputil` instead?

File: pkg/auth/authenticator/password/ldappassword/ldaputil/getattribute.go (hunk 1 above)
Comment (stevekuznetsov) [inline pkg/auth/authenticator/password/ldappassword/ldaputil/getattribute.go]: This is used in the `ldaputil` package but also in quite a few of the LDAP-interacting portions of the LDAP sync. Where is it better off living?

**Comment by stevekuznetsov** [comment]
If we wanted to migrate the utility functions and structs used in LDAP sync into this package it would also make much more sense for it to be somewhere else, although the LDAP sync has nothing to do with `auth`.

**Comment by deads2k** [comment]
you can expose `GetAttributeValue` from `identityfactory.go`

**Comment by deads2k** [comment]
This looks like a straight move in preparation for ldap groups sync.  Exposing methods can be done when they're needed in the next pull.  

@liggitt can remove the tag if he wants time to review.  [merge]

**Comment by liggitt** [comment]
```
FAIL	github.com/openshift/origin/pkg/auth/ldaputil [build failed]
```

**Comment by stevekuznetsov** [comment]
@liggitt Fixed.

**Comment by liggitt** [comment]
[merge]ing on @deads2k's recognizance

File: pkg/cmd/server/api/v1/types_test.go [hunk 2]
```diff
@@ -166,7 +166,7 @@ oauthConfig:
     name: ""
     provider:
       apiVersion: v1
-      attributes:
+      attributeMappings:
```
Comment (liggitt) [inline pkg/cmd/server/api/v1/types_test.go]: the serialized field should not have been renamed... need to revert

**Pull request merged**